Mathlib/NumberTheory/TsumDivsorsAntidiagonal.lean

```diff
@@ -49,3 +56,78 @@ lemma sigmaAntidiagonalEquivProd_symm_apply_fst (x : ℕ+ × ℕ+) :
 
 lemma sigmaAntidiagonalEquivProd_symm_apply_snd (x : ℕ+ × ℕ+) :
     (sigmaAntidiagonalEquivProd.symm x).2 = (x.1.1, x.2.1) := rfl
+
+section tsum
+
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜] [CompleteSpace 𝕜] [NormSMulClass ℤ 𝕜]
+
+omit [NormSMulClass ℤ 𝕜] in
+lemma summable_norm_pow_mul_geometric_div_one_sub (k : ℕ) {r : 𝕜} (hr : ‖r‖ < 1) :
+    Summable fun n : ℕ ↦ n ^ k * r ^ n / (1 - r ^ n) := by
+  simp only [div_eq_mul_one_div ( _ * _ ^ _)]
+  apply summable_mul_tendsto_const (c := 1 / (1 - 0))
+    (by simpa using summable_norm_pow_mul_geometric_of_norm_lt_one k hr)
+  simpa only [Nat.cofinite_eq_atTop] using
+   tendsto_const_nhds.div ((tendsto_pow_atTop_nhds_zero_of_norm_lt_one hr).const_sub 1) (by simp)
+
+theorem summable_divisorsAntidiagonal_aux (k : ℕ) {r : 𝕜} (hr : ‖r‖ < 1) :
```

Comment:
The statement of this lemma (with `c.2.1.2` etc) is a bit off-putting. Do you intend to re-use it other than in `summable_prod_mul_pow`? If not, please make it private or inline it in the proof of that lemma (with a `suffices` perhaps).